TSOClient/tso.content/framework/FAR1Provider.cs

```diff
@@ -189,29 +189,36 @@ public void Init()
                 }
                 FarFiles = farFiles.ToArray();
             }
-
+            
             foreach (var farPath in FarFiles){
                 var archive = new FAR1Archive((TS1)?Path.Combine(ContentManager.TS1BasePath, farPath):ContentManager.GetPath(farPath), !TS1);
                 var entries = archive.GetAllFarEntries();
 
                 foreach (var entry in entries)
                 {
-                    var referenceItem = new Far1ProviderEntry<T>(this)
-                    {
-                        Archive = archive,
-                        FarEntry = entry
-                    };
-                    if (entry.Filename != null)
+                    try
                     {
-                        EntriesByName[entry.Filename] = referenceItem;
-                        var ext = Path.GetExtension(entry.Filename).ToLowerInvariant();
-                        List<Far1ProviderEntry<T>> group = null;
-                        if (!EntriesOfType.TryGetValue(ext, out group))
+                        var referenceItem = new Far1ProviderEntry<T>(this)
+                        {
+                            Archive = archive,
+                            FarEntry = entry
+                        };
+                        if (entry.Filename != null)
                         {
-                            group = new List<Far1ProviderEntry<T>>();
-                            EntriesOfType[ext] = group;
+                            EntriesByName[entry.Filename] = referenceItem;
+                            var ext = Path.GetExtension(entry.Filename).ToLowerInvariant();
+                            List<Far1ProviderEntry<T>> group = null;
+                            if (!EntriesOfType.TryGetValue(ext, out group))
+                            {
+                                group = new List<Far1ProviderEntry<T>>();
+                                EntriesOfType[ext] = group;
+                            }
+                            group.Add(referenceItem);
                         }
-                        group.Add(referenceItem);
+                    }
+                    catch (Exception exception)
+                    {
+                        throw new Exception($"The FAR archive at: {archive} adding entry: {entry.Filename} threw the following exception: ", exception);
```

Comment:
This doesn't really have anything to do with the PR. What could even throw in here?